### ISSUE TITLE
Taskbar at top has problems with window snapping / maximisation.

### ISSUE THREAD

**Issue by xSimone** [valinet/ExplorerPatcher#3045]
Yes, you guessed it, I am the most hated, a heathen if you will, a disgusting figure of the lowest life on earth..

.... a Top-Screen Taskbar user.

The problem is sometimes when maximising a window, the top of the window clips under the task bar and it's difficult to move it back down, while there's a gap at the bottom where the bottom taskbar would be.

This I believe happens mostly when gaming, though I'll gather more information and screenshots soon and post them here.

**Comment by pyrates999** [comment]
Are you using the windows 10 taskbar or windows 11 taskbar?

Please post windows version and build.  You can find this by running winver.
Please post explorer patcher version.

Please include what settings you changed in explorer patcher properties application.  These should be screen shots.

Please include any other software you have installed that modifies the UI of windows.

**Comment by Amrsatrio** [comment]
Usually restarting Explorer fixes this issue, other than that, I don't know how to make this not happen again...

**Comment by Firav** [comment]
Heyo! Another top taskbar user here...
@Amrsatrio were you ever able to find a workaround for this? Other than restarting explorer?

**Comment by Amrsatrio** [comment]
No. I need to explore the unexplored territory (and also hell) in explorer.exe to figure out why this is happening. Safe to say this is a Windows bug that we need to patch ourselves to fix it.

**Comment by Firav** [comment]
I've enabled this setting in the Advanced section and it seems to reset it, rather than restarting explorer completely. Absolutely just a quickhand workaround on my part.

![Image](https://github.com/user-attachments/assets/4ad258bc-5d77-422c-a83c-589b1d39d8de)

**Comment by xSimone** [comment]
> Are you using the windows 10 taskbar or windows 11 taskbar?
> 
> Please post windows version and build. You can find this by running winver. Please post explorer patcher version.
> 
> Please include what settings you changed in explorer patcher properties application. These should be screen shots.
> 
> Please include any other software you have installed that modifies the UI of windows.


Jesus christ it has been a year, I cannot believe I didn't get notifications for this.
I think I made a second post because I swear I've come back to talk about this issue.

Win10 
Latest version as of the time posted for both. I'm always up to date.
Nothing other than moving the taskbar to the top.
Nothing would be modifying the UI other than ExplorerPatcher, unless Mega counts somehow given it's a Cloud service.

**Comment by xSimone** [comment]
> Heyo! Another top taskbar user here... [@Amrsatrio](https://github.com/Amrsatrio) were you ever able to find a workaround for this? Other than restarting explorer?

Hi,

So my current fix is keeping TaskBar unlocked, and then I just clickhold the edge, and slightly move it down, it snaps back into place and is normal again.

Most windows then properly snap again too.. sometimes not.

**Comment by pyrates999** [comment]
Please post your build of windows 10.  You can find out by running **winver**.  Saying it's the latest one doesn't mean anything as people have said that on here and they hadn't installed windows updates for months.

And since you're on windows 10, you can verify if this is a bug with EP or not.  Uninstall EP, and reboot.  Then make sure your taskbar is still at the top and try to cause this.

**Comment by xSimone** [comment]
> Please post your build of windows 10. You can find out by running **winver**. Saying it's the latest one doesn't mean anything as people have said that on here and they hadn't installed windows updates for months.
> 
> And since you're on windows 10, you can verify if this is a bug with EP or not. Uninstall EP, and reboot. Then make sure your taskbar is still at the top and try to cause this.

I'm on Windows 11.

26100.3624

**Comment by pyrates999** [comment]
that's a preview update: https://support.microsoft.com/en-us/topic/march-27-2025-kb5053656-os-build-26100-3624-preview-4c35f1c4-1ae6-41ef-a317-3d8ee2e73975

In the settings app, untick "Get the latest updates as soon as available" so that preview updates are not automatically installed.

All preview updates are then shown as download and install and are never automatically installed if you have the above unticked.

Preview updates are also not fully tested by Microsoft and thus could have bugs.

Since you're on windows 11, then you can't do the test I suggested.

**Comment by xSimone** [comment]
> that's a preview update: https://support.microsoft.com/en-us/topic/march-27-2025-kb5053656-os-build-26100-3624-preview-4c35f1c4-1ae6-41ef-a317-3d8ee2e73975
> 
> In the settings app, untick "Get the latest updates as soon as available" so that preview updates are not automatically installed.
> 
> All preview updates are then shown as download and install and are never automatically installed if you have the above unticked.
> 
> Preview updates are also not fully tested by Microsoft and thus could have bugs.
> 
> Since you're on windows 11, then you can't do the test I suggested.

Ah I forgot to uncheck it when I reinstalled Windows a month ago. 

Whoops.

I unchecked it but yeah. 

The bug isn't too bad, just the snapping can be annoying at times. but it's easy to fix by clicking the resize edge and giving it a twitch.

**Comment by astro46** [comment]
I installed EP 3 days ago.  Each day (3 times today, so far) I've had to restart EP to see tops of windows below the taskbar, which is on top of screen, (the reason I use EP).
Is there a setting to modify to keep tops of windows from hiding under the top taskbar?

**Comment by xSimone** [comment]
Unfortunately issue persists, especially with Steam.

**Comment by astro46** [comment]
I have discovered that setting "properties>advanced>double click taskbar to toggle auto-hide"  to on, allows me to double click taskbar and get the tops of windows back in view, without restarting EP.    This is more convenient,  and perhaps it may help to indicate a direction for fixing the issue.

**Comment by manuelaigner** [comment]
I have exactly the same issue. 
It seems to me it happens if your display switches in a different mode. At my configuration it reproducible happens when the main display (I've 3x1440p screens) switches from SDR to HDR due to a HDR game startup.

**Comment by astro46** [comment]
for me,  no display switching to different mode, that I am aware of.  no game startups. I haven't noticed a correlation with anything. 

edit: I am seeing that hiding tops of windows occurs when coming out of sleep mode. Not all of the time. Sometimes.

**Comment by xSimone** [comment]
> I have exactly the same issue. 
> It seems to me it happens if your display switches in a different mode. At my configuration it reproducible happens when the main display (I've 3x1440p screens) switches from SDR to HDR due to a HDR game startup.

Yes that makes a lot of sense. I believe that's what happens to me too. 

My laptop is 16:10 and I have HDR that comes on depending what game I launch / full screen.

**Comment by manuelaigner** [comment]
I also can confirm @astro46 's behavior. Sometimes it happens after system startup, or when resuming after standby.

**Comment by salvattore001** [comment]
hi. i have a similar issue with on-top taskbar with auto hide.
after sleep or display power off the taskbar doesn't appear when hovering. 
it seems to be behind the maximized windows.

**Comment by astro46** [comment]
I have discovered that the problem with my method of getting tops of windows back, mentioned above (setting to double click to hide taskbar, then using double clicks, several, to get top of windows back),  is that if the task bar is filled with apps, then there is no where to click.  have to close apps to get space to double click to fix the problem.

**Comment by orthoxerox** [comment]
Windows 11: 26200.7623

Explorer Patcher: 26100.4946.69.6

I noticed that this happened to me when GHelper was running with the refresh rate set to "auto" and I plugged my laptop in or out. I debugged the code and found out that the work area is changed when it calls `ChangeDisplaySettingsEx(laptopScreen, ref dm, IntPtr.Zero, DisplaySettingsFlags.CDS_UPDATEREGISTRY, IntPtr.Zero)` from `user32.dll`. I replaced `CDS_UPDATEREGISTRY` with `0` but achieved the same result.

I also tried changing the refresh rate manually in Windows settings and reproduced the same issue. Changing the resolution *doesn't* trigger this.

**Comment by Amrsatrio** [comment]
Hi, thank you very much for the repro. Win+Fn+R on my Lenovo Slim 7x reproduces this issue all the time and flawlessly.

Looks like the culprit is in the work area handler which resides in explorer.exe, and not something EP has full control over. Might take a while to find which explorer.exe version started shipping with this quirk and design the patch.

I'll let you know once I've brewed a solution to this. Thanks!

**Comment by Amrsatrio** [comment]
Seems like an internal display having switchable refresh rates is a hard requirement to repro this issue...

**Comment by xSimone** [comment]
> Seems like an internal display having switchable refresh rates is a hard requirement to repro this issue...

Hi, that's most likely the trigger for me. When entering a fullscreen game if it has an FPS lock that probably does cause the issue.

Changing my refresh rate manually through a tool I like (G-Helper) is an easy way to reproduce this.

<img width="683" height="134" alt="Image" src="https://github.com/user-attachments/assets/dca9ec1d-32e2-40af-bf58-e65eeff8659d" />

**Comment by orthoxerox** [comment]
@Amrsatrio Changing the orientation of the screen on a tablet PC is another way to trigger this issue. I've just tilted mine to shake off a cookie crumble and triggered it.

**Comment by astro46** [comment]
EP has gotten worse in this regard:  apparently it upgraded, now I can no longer fix the problem by clicking a few times on the tax bar.  Now have to right click>properties>restart EP.  Sad.  
I was hoping the problem would be fixed, not made worse.
